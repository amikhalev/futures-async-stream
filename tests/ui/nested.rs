// compile-fail

#![feature(generators)]

use futures_async_stream::async_stream;

#[async_stream(item = i32)]
async fn stream(x: i32) {
    for i in 1..=x {
        yield i
    }
}

#[async_stream(item = i32)]
async fn _stream1() {
    async {
        #[for_await]
        for i in stream(2) {
            yield i * i; //~ ERROR `async` generators are not yet supported [E0727]
        }
    }
        .await;
}

fn main() {}
