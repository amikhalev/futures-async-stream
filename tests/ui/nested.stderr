error[E0727]: `async` generators are not yet supported
  --> $DIR/nested.rs:11:13
   |
11 |             yield i * i; //~ ERROR `async` generators are not yet supported [E0727]
   |             ^^^^^^^^^^^

error: aborting due to previous error

